src/main/java/org/wildfly/security/mechanism/oauth2/OAuth2Client.java

```diff
@@ -62,12 +62,12 @@ public OAuth2InitialClientMessage getInitialResponse() throws AuthenticationMech
         assertTrue(credentialCallback.isCredentialTypeSupported(BearerTokenCredential.class));
 
         BearerTokenCredential credential = credentialCallback.getCredential(BearerTokenCredential.class);
-        final String token = credential.getToken();
 
-        if (token == null) {
```

Comment:
You don't have to change this, but one cool trick you can do is:

```final String token = credentialCallback.applyToCredential(BearerTokenCredential.class, Credential::getToken);```

and save a couple lines of code; then you only have to check if token is null.